src/PointwiseFunctions/Hydro/EquationsOfState/Tabulated3d.cpp

```diff
@@ -18,6 +18,68 @@ EQUATION_OF_STATE_MEMBER_DEFINITIONS(template <bool IsRelativistic>,
                                      Tabulated3D<IsRelativistic>, double, 3)
 EQUATION_OF_STATE_MEMBER_DEFINITIONS(template <bool IsRelativistic>,
                                      Tabulated3D<IsRelativistic>, DataVector, 3)
+
+template <bool IsRelativistic>
+template <class DataType>
+Scalar<DataType> Tabulated3D<IsRelativistic>::
+    equilibrium_electron_fraction_from_density_temperature_impl(
+        const Scalar<DataType>& rest_mass_density,
+        const Scalar<DataType>& temperature) const {
+  Scalar<DataType> electron_fraction = make_with_value<Scalar<DataType>>(
+      rest_mass_density, electron_fraction_lower_bound());
+
+  Scalar<DataType> converted_electron_fraction;
+  Scalar<DataType> log_rest_mass_density;
+  Scalar<DataType> log_temperature;
+
+  convert_to_table_quantities(
+      make_not_null(&converted_electron_fraction),
+      make_not_null(&log_rest_mass_density), make_not_null(&log_temperature),
+      electron_fraction, rest_mass_density, temperature);
+
+  if constexpr (std::is_same_v<DataType, double>) {
+    const auto& log_rho = get(log_rest_mass_density);
+    const auto& log_T = get(log_temperature);
+
+    const auto f = [this, log_rho, log_T](const double ye) {
```

Comment:
Just to make sure I understand, this is finding the location where `DeltaMu` is zero?

Reply:
That's right. This fixes the free-streaming beta-equilibrium electron fraction. I added a comment.

Reply:
Thanks!